frappe/public/js/frappe/form/controls/currency.js

```diff
@@ -1,21 +1,6 @@
 frappe.ui.form.ControlCurrency = frappe.ui.form.ControlFloat.extend({
-	eval_expression: function(value) {
-		if (typeof value ==='string' && value.match(/^[0-9+-/* ]+$/)) {
-			// Removes seperator
-			value = strip_number_groups(value, this.get_number_format());
-
-			try {
-				return eval(value);
-			} catch (e) {
-				return value;
-			}
-		}
-		// If not string
-		return value;
-	},
-
 	format_for_input: function(value) {
-		var formatted_value = format_number(parseFloat(value), this.get_number_format(), this.get_precision());
```

Comment:
Problem: When a CSV file is pasted to a table, format_for_input is called before the string value (eg. "44,444.01") is striped out of its separators. So because of parseFloat("44,444.01") it gets passed as 44.00 into format_number()

Solution: format_number() parses value correctly even if string value is passed.